cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql

```diff
@@ -79,38 +84,24 @@ predicate isNonConst(DataFlow::Node node, boolean isIndirect) {
       )
     )
     or
-    exists(Parameter p | p = e.(VariableAccess).getTarget() |
-      p.getFunction().getName() = "main" and p.getType() instanceof PointerType
-    )
-    or
-    e instanceof CrementOperation
-    or
-    e instanceof AddressOfExpr
-    or
-    e instanceof ReferenceToExpr
-    or
-    e instanceof AssignPointerAddExpr
-    or
-    e instanceof AssignPointerSubExpr
-    or
-    e instanceof PointerArithmeticOperation
```

Comment:
This change (and a few similar cases) appears to explain some of the results we lose in test cases. It seems we were quite keen on writing test cases with pointer arithmetic. I guess in the real world we get a lot of false positives caused by these as well (hence why you want to delete them)?

Does converting `NonConstFlowConfig` from data flow to taint put us in a better (or worse) place?

Reply:
Yea, any pointer arithmetic on a constant string, is a constant string. Can the arithmetic go out of bounds? That's a separate concern. All the regressions in the unit test that disappear I would say were false positives (in fact, one was even marked as such).

Reply:
> All the regressions in the unit test that disappear I would say were false positives (in fact, one was even marked as such).

Ah, I see what you mean.  We want (and most likely have) flow *through* most of these constructs but we don't want them to be sources themselves.  The tests are in fact testing the latter.

I'm happy for you to update the test labels from `// BAD` to `// GOOD` in these cases.